File: khalti-android/src/main/java/khalti/checkOut/EBanking/deepLinkReceiver/DeepLinkContract.java
package khalti.checkOut.EBanking.deepLinkReceiver;

import java.util.HashMap;

import khalti.checkOut.api.Config;

interface DeepLinkContract {
    interface View {
        HashMap<String, Object> receiveEBankingData();

        Config getConfigFromFile();

        void closeDeepLink();

        void setListener(Listener listener);
    }

    interface Listener {
        void receiveEBankingData();
    }
}

File: khalti-android/src/main/java/com/khalti/checkout/banking/deepLinkReceiver/DeepLinkContract.kt
package com.khalti.checkout.banking.deepLinkReceiver

import com.khalti.base.LifeCycle
import com.khalti.checkout.helper.Config

interface DeepLinkContract {
    interface View {

        val configFromFile: Config

        fun receiveEBankingData(): Map<String, Any>?

        fun closeDeepLink()

        fun setPresenter(presenter: Presenter)
    }

    interface Presenter : LifeCycle
}
